crates/sui-json-rpc-types/src/sui_event.rs

```diff
@@ -122,6 +124,22 @@ impl SuiEvent {
     }
 }
 
+impl Display for SuiEvent {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let output = format!(
+            " ┌──\n │ EventID: {}\n │ PackageID: {}\n │ Transaction Module: {}\n │ Sender: {} \n │ EventType: {}:{}\n │ ParsedJSON: {}\n │ BCS: {:?}\n",
+            self.id.tx_digest, self.id.event_seq, self.package_id, self.transaction_module, self.sender, self.type_, self.parsed_json, self.bcs);
```

Comment:
Do we want the JSON field?
Do we want the BCS field? If yes, when transformed to JSON this is encoded as Base58, so probably we want to do the same?

Reply:
Re: the JSON field, could we display it using `tabled!`?  Re: `bcs`, given that the display format is for people to read, I don't think we necessarily need it, but I'll leave the final call up to you.  If you do leave it though, it should be encoded as Base64, not Base58.

Reply:
I am curious about this Base64 vs Base58 stuff. My understanding was that previously, when the bcs field was displayed it was encoded as Base58: https://github.com/MystenLabs/sui/blob/main/crates/sui-json-rpc-types/src/sui_event.rs#L47-L50. What am I missing here?

Reply:
Oh interesting -- that's a bit of an outlier, it may have come from a time when we were thinking of only using `Base58` everywhere, but that was very short-lived because that comes at an efficiency cost (64/58 size blow-up).

So in practice, we now use `Base58` for small binary blobs where it's bad to confuse similar characters etc, i.e. digests and other similar hashes, and we use `Base64` for other bigger binary blobs (like bcs data).

Reply:
Thank you for the context. If I am understanding you right, I should use `Base64` for pretty printing, but what happens when the user requests the `json` output? The two values will be different and it will perhaps confuse folks? Do we want to update the encoding scheme of the field?

Reply:
- Changing the format for Event will indeed be a breaking change, so we can't just do that. 
- For pretty printing, I don't think we should print large binary blobs -- I'm not sure that's helpful.
- If people request JSON, then we should give the format that the RPC outputs.  This *should* be Base64, but if it's Base58 at the moment, so be it, we should stay consistent, and make the change starting at the RPC, with a breaking change (e.g. when we adopt GraphQL)

Reply:
Cool, that makes sense. I've been thinking if it's really useful to have these fields, so I think we can go with a tradeoff here: remove both parsed json and bcs from the normal display, but have them available when the --json flag is passed.

Reply:
To be clear -- these fields should of course exist if `--json` is passed, because that is a machine readable output, and I never meant to put that in question.

I think it's also fine (useful, even) to have the parsed JSON in the pretty, human readable output, because it is quite meaningful to what the event is.